build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -1,60 +1,69 @@
+    <!-- Current Runtime Information -->
+    <Rid>
+      <Output TaskParameter="OutputRid" PropertyName="Rid" />
+    </Rid>
+    <Architecture>
+      <Output TaskParameter="OutputArchitecture" PropertyName="Architecture" />
+    </Architecture>
+
+    <!-- Common Properties -->
     <PropertyGroup>
       <RepoRoot>$(MSBuildThisFileDirectory)/..</RepoRoot>
-      <OutputDirectory>$(RepoRoot)/artifacts/win10-x64/stage2</OutputDirectory>
-      <ProductName>dotnet</ProductName>
-      <CommitCount>1000</CommitCount>
-      <SdkVersion>1.0.0-preview3-$(CommitCount)</SdkVersion>
-      <IntermediateDirectory>$(MSBuildThisFileDirectory)/../artifacts/win10-x64/intermediate</IntermediateDirectory>
-      <BaseOutputDirectory>$(MSBuildThisFileDirectory)/../artifacts/win10-x64</BaseOutputDirectory>
-      <Architecture>x64</Architecture>
-      <BranchInfo>$(RepoRoot)/branchinfo.txt</BranchInfo>
+      <BaseOutputDirectory>$(MSBuildThisFileDirectory)/../artifacts/$(Rid)</BaseOutputDirectory>
+      <OutputDirectory>$(BaseOutputDirectory)/stage2</OutputDirectory>
+      <Stage2CompilationDirectory>$(BaseOutputDirectory)/stage2compilation</Stage2CompilationDirectory>
+      <IntermediateDirectory>$(BaseOutputDirectory)/intermediate</IntermediateDirectory>
+      <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     </PropertyGroup>
 
-    
+    <!-- Generates BuildVersion Item
+        BuildVersion Metadata:
+          - Major
+          - Minor
+          - Patch
+          - ReleaseSuffix
+          - CommitHash
+          - CommitCount
+          - VersionSuffix
+          - SimpleVersion
+          - NugetVersion
+          - MsiVersion            -->
+    <GenerateBuildVersionInfo RepoRoot="$(RepoRoot)">
+        <Output TaskParameter="OutputBuildVersionInfo"
+                ItemName="BuildVersion" />
+    </GenerateBuildVersionInfo>
+
+    <PropertyGroup>
+      <SdkVersion>@(BuildVersion ->'%(NugetVersion)')</SdkVersion>
+    </PropertyGroup>
 
     <!-- Initial Artifact State Generation -->
     <ItemGroup>
-      <ArtifactNames Include="None">
+      <ArtifactNames Include="ArtifactNames">
         <Sdk>dotnet-sdk</Sdk>
         <SdkDebug>dotnet-sdk-debug</SdkDebug>
         <CombinedHostHostFxrFrameworkSdk>dotnet-dev</CombinedHostHostFxrFrameworkSdk>
         <CombinedFrameworkSdk>dotnet-sharedframework-sdk</CombinedFrameworkSdk>
       </ArtifactNames>
 
-      <ArtifactNamesWithVersions Include="None">
-        <Sdk>@(ArtifactNames -> '%(Sdk)-$(SdkVersion)')</Sdk>
-        <SdkDebug>@(ArtifactNames -> '%(SdkDebug)-$(SdkVersion)')</SdkDebug>
-        <CombinedHostHostFxrFrameworkSdk>@(ArtifactNames -> '%(CombinedHostHostFxrFrameworkSdk)-$(SdkVersion)')</CombinedHostHostFxrFrameworkSdk>
-        <CombinedFrameworkSdk>@(ArtifactNames -> '%(CombinedFrameworkSdk)-$(SdkVersion)')</CombinedFrameworkSdk>
+      <ArtifactNamesWithVersions Include="ArtifactNamesWithVersions">
+        <aSdk>dotnet-sdk-$(SdkVersion)</aSdk>
```

Comment:
I'm wondering if these should just be properties, and not using an Item to group these properties. It feels weird to me because:
1. There aren't "many" of these items, just one.
2. The fact that we have "Include" with the same name as the XML element, kind of goes to show that there should only be a single one of these, and makes them feel more like properties to me.
(I'm no MSBuild expert, just wondering what the best practice is in situations like this.)

@weshaggard - my resident "MSBuild expert" to get his thoughts on this pattern.